### PR TITLE
Mark 403 (Forbidden) for fetching turn credentials as terminal error and avoid retrying.

### PR DESCRIPTION
**Pull request by anuranduttaroy**
**Issue #:**  NA

**Description of changes:**
Mark 403 (Forbidden) for fetching turn credentials as terminal error and avoid retrying.

**Testing**

1. Have you successfully run `npm run build:release` locally?
Yes.

2. How did you test these changes?
Tested it manually & added unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/meetingsession/MeetingSessionStatusCode.ts [hunk 1]
```diff
@@ -118,6 +118,11 @@ export enum MeetingSessionStatusCode {
    * Session update produces incompatible SDP.
    */
   IncompatibleSDP = 21,
+
+  /**
+   * This can happen when you attempt to join a meeting which has ended
+   */
+  ForbiddenToFetchTrunCredentials = 22,
```
Comment (aburnson) [inline src/meetingsession/MeetingSessionStatusCode.ts:125]: s/Trun/Turn/

File: src/task/ReceiveTURNCredentialsTask.ts [hunk 2]
```diff
@@ -62,6 +63,12 @@ export default class ReceiveTURNCredentialsTask extends BaseTask {
       try {
         const responseBody = await fetch(Versioning.urlWithVersion(this.url), options);
         this.context.logger.info(`received TURN credentials`);
+        if (responseBody.status && responseBody.status === 403) {
+          let error = new Error(
+            `Forbidden to fetch TURN Credentials due to the meeting status code: ${MeetingSessionStatusCode.ForbiddenToFetchTrunCredentials}`
```
Comment (aburnson) [inline src/task/ReceiveTURNCredentialsTask.ts:68]: This string is the only place where this error code is used, so why do we need it?

File: src/meetingsession/MeetingSessionStatusCode.ts (hunk 1 above)
Comment (anuranduttaroy) [reply src/meetingsession/MeetingSessionStatusCode.ts]: Fixed.

File: src/task/ReceiveTURNCredentialsTask.ts (hunk 2 above)
Comment (anuranduttaroy) [reply src/task/ReceiveTURNCredentialsTask.ts]: Do you suggest using an existing error code for this case?

Comment (aburnson) [reply src/task/ReceiveTURNCredentialsTask.ts]: As far as I can tell, the error code is not set or returned anywhere. If no one can actually consume this error code (other than by parsing it out of a string—which could change), then what is the purpose of having an error code?

File: test/dommock/DOMMockBuilder.ts [hunk 3]
```diff
@@ -759,7 +759,11 @@ export default class DOMMockBuilder {
         if (mockBehavior.responseSuccess) {
```
Comment (simmkyu) [inline test/dommock/DOMMockBuilder.ts:759]: Could you simplify this `status()` getter by removing `mockBehavior.responseSuccess ` and returning your `mockBehavior.responseStatusCode`? It requires small changes in `MeetingSessionPOSTLogger.test.ts`.

```
get status(): number {
  return mockBehavior.responseStatusCode;
}
```

File: src/task/ReceiveTURNCredentialsTask.ts [hunk 4]
```diff
@@ -62,6 +63,14 @@ export default class ReceiveTURNCredentialsTask extends BaseTask {
       try {
         const responseBody = await fetch(Versioning.urlWithVersion(this.url), options);
         this.context.logger.info(`received TURN credentials`);
+        if (responseBody.status && responseBody.status === 403) {
+          let error = new Error(
```
[the remark this answers is not in the document]
Comment (anuranduttaroy) [reply src/task/ReceiveTURNCredentialsTask.ts]: Updated

File: src/meetingsession/MeetingSessionStatusCode.ts [hunk 5]
```diff
@@ -118,6 +118,11 @@ export enum MeetingSessionStatusCode {
    * Session update produces incompatible SDP.
    */
   IncompatibleSDP = 21,
+
+  /**
+   * This can happen when you attempt to join a meeting which has ended
+   */
+  ForbiddenToFetchTurnCredentials = 22,
```
[the remark this answers is not in the document]
Comment (anuranduttaroy) [reply src/meetingsession/MeetingSessionStatusCode.ts]: Updated

File: test/dommock/DOMMockBuilder.ts (hunk 3 above)
Comment (anuranduttaroy) [reply test/dommock/DOMMockBuilder.ts]: Updated

**Comment by simmkyu** [comment]
LGTM. In a builder's app, `AudioVideoObserver.audioVideoDidStop` will be called with a new meeting status code `TURNCredentialsForbidden`. They can handle this code accordingly, e.g. showing the error UI.

**Pull request merged**